### PR TITLE
built in collision detection system

### PR DESCRIPTION
**Pull request by YunHsiao**
* more schema params supported in cannon mode
* built-in collision detection system & test scene
* add mobile input support for navigation scene
* simplify system removes, no need to do explicitly loops
* removed dt interface for physics system, too unstable

### CODE REVIEW

**Comment by YunHsiao** [comment]
for #692

File: lib/framework/physics/built-in-body.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    // create shape
+    this.shapeType = typeMap[options.type];
+    this._shapeModelSpace = this.createShape(options);
+    this.shape = this.createShape(options);
+    this._system._fullSimulation = true;
+    this.collisionFilterGroup = 1;
+    this.collisionFilterMask  = 1;
+  }
+  
+  createShape(options) {
```
[the remark this answers is not in the document]
Comment (YunHsiao) [reply lib/framework/physics/built-in-body.js]: nope, still need center and size info from options

File: lib/framework/physics/built-in-world.js [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
+        let bodyB = this._cols.data[j];
+
+        if( (bodyA.collisionFilterGroup & bodyB.collisionFilterMask) === 0 ||
+          (bodyB.collisionFilterGroup & bodyA.collisionFilterMask) === 0) {
+          continue;
+        }
+
+        if (bodyA.intersect(bodyB)) {
+          event.body = bodyA; event.target = bodyB;
+          bodyA._entity.emit('collide', event);
```
[the remark this answers is not in the document]
Comment (YunHsiao) [reply lib/framework/physics/built-in-world.js]: yes, but that would mean... to create a new object for every collision? is it performance-wise acceptable?

File: lib/framework/physics/cannon-body.js [hunk 3]
```diff
@@ -91,27 +114,45 @@ export default class CannonCollider extends CANNON.Body {
...
     if (this.shapeType != 'sphere') return;
+    if (!val) val = this._radius;
     this._entity.getWorldScale(v3_tmp);
     this.shape.radius = val * maxComponent(v3_tmp);
     this.shape.updateBoundingSphereRadius();
     this.updateBoundingRadius();
+    this._radius = val;
+  }
+
+  setUpdateMode(In, Out) {
```
[the remark this answers is not in the document]
Comment (YunHsiao) [reply lib/framework/physics/cannon-body.js]: nope, `in` is a javascript keyword. but still, should have followed camelCase.

File: lib/framework/physics/built-in-body.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dabingnn) [reply lib/framework/physics/built-in-body.js]: got taht

File: lib/framework/physics/cannon-body.js [hunk 4]
```diff
@@ -91,27 +114,45 @@ export default class CannonCollider extends CANNON.Body {
...
-  manualUpdate() {
-    this.updateIn();
-    this.setSize();
+  manualUpdate(fixedRotation) {
+    this.updateIn(null, fixedRotation);
+    this.setSize(); this.setRadius();
   }
 
-  updateIn() {
+  updateIn(event, fixedRotation = this.fixedRotation) {
```
[the remark this answers is not in the document]
Comment (YunHsiao) [reply lib/framework/physics/cannon-body.js]: the function is meant to be called from both collision events and manual updates.
when function as event callbacks, the parameter is always undefined and automatically defaults to `this.fixedRotation` to keep align with its state. 
when called manually, the user should be able to decide whether to update rotation or not.

File: lib/framework/physics/built-in-world.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dabingnn) [reply lib/framework/physics/built-in-world.js]: so we need to add a ticket for this, for now, just force the event to be consumed immediately.

**Comment by dabingnn** [comment]
https://github.com/cocos-creator/engine-3d/issues/556

**Pull request merged**